### PR TITLE
Add org docs

### PR DESCRIPTION
**Pull request by dataclouder**
##  Add documentation for Org resource

* Add missing property to org resource
* Add definition for non self-explanatory resources
* Remove "use_server_boot_sequence" property, as the docs define it as "This value is ignored."
* Add missing error handling from several resources
* Update vendor directory

### CODE REVIEW

File: vcd/resource_vcd_org_test.go [hunk 1]
```diff
@@ -77,9 +80,20 @@ func testAccCheckOrgDestroy(s *terraform.State) error {
 			continue
 		}
 
-		org, err := govcd.GetOrgByName(conn.VCDClient, rs.Primary.Attributes["name"])
-		if org != (govcd.Org{}) || err != nil {
-			return fmt.Errorf("org with name %s was found", rs.Primary.Attributes["name"])
+		deleted := false
+		var org govcd.Org
+		var err error
+		for N := 0; N < 10; N++ {
```
Comment (vbauzysvmware) [inline vcd/resource_vcd_org_test.go]: I would say this isn't sustainable for all systems. If we need to wait 100 count? Couldn't we use while?

File: website/docs/r/org.html.markdown [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Org name
+* `full_name` - (Required) Org full name
+* `delete_recursive` - (Required) - pass delete_recursive=true as query parameter to remove an organization or VDC and any objects it contains that are in a state that normally allows removal.
+* `delete_force` - (Required) - pass delete_force=true and delete_recursive=true to remove an organization or VDC and 
+   any objects it contains, regardless of their state.
+* `is_enabled` - (Optional) - True if this organization is enabled (allows login and all other operations).
```
Comment (vbauzysvmware) [inline website/docs/r/org.html.markdown]: Mising default value

File: vcd/resource_vcd_org_test.go (hunk 1 above)
Comment (dataclouder) [reply vcd/resource_vcd_org_test.go]: Right. I was actually fishing for comments on this one, as I have seen similar methods in a few places.
But we need to define a few public timeout values for long term operations and then use them consistently. I have a related task in my TODO list.

**Comment by dataclouder** [comment]
The use of "_" is idiomatic when we want to explicitly ignore errors, but at the same time we want to silence detection tools. If there is a particular error that we want to handle for `FPrintf`, please suggest it.

File: vcd/resource_vcd_org_test.go (hunk 1 above)
Comment (vbauzysvmware) [reply vcd/resource_vcd_org_test.go]: ok

**Comment by vbauzysvmware** [comment]
Yes, but I don't see difference between _ = func and func .

**Comment by dataclouder** [comment]
There is no substantial difference. The end result is that we don't check the error. 
However the `_ = func` syntax will signal the tools like [errcheck](https://github.com/kisielk/errcheck) (which we use in our repo) that we take responsibility for the lack of error checking, and therefore we don't get notified for this call.

**Comment by vbauzysvmware** [comment]
ok. Got it.

File: vcd/resource_vcd_org.go [hunk 3]
```diff
@@ -34,44 +39,53 @@ func resourceOrg() *schema.Resource {
...
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
 				ForceNew: false,
 			},
+			"is_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     false,
```
Comment (lvirbalas) [inline vcd/resource_vcd_org.go]: Shouln't organisation be enabled by default?

File: vcd/resource_vcd_org.go [hunk 4]
```diff
@@ -34,44 +39,53 @@ func resourceOrg() *schema.Resource {
...
+				Description: "True if this organization is enabled (allows login and all other operations).",
+			},
 			"deployed_vm_quota": &schema.Schema{
-				Type:     schema.TypeInt,
-				Optional: true,
-				Default:  -1,
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     -1,
+				Description: "Maximum number of virtual machines that can be deployed simultaneously by a member of this organization.",
```
Comment (lvirbalas) [inline vcd/resource_vcd_org.go]: Thanks for adding these descriptions to arguments!

File: vcd/resource_vcd_org_test.go [hunk 5]
```diff
@@ -77,9 +80,20 @@ func testAccCheckOrgDestroy(s *terraform.State) error {
...
+			org, err = govcd.GetOrgByName(conn.VCDClient, rs.Primary.Attributes["name"])
+			deleted = (org == (govcd.Org{})) && (err != nil)
+			if deleted {
+				break
+			}
+			time.Sleep(time.Second)
```
Comment (lvirbalas) [inline vcd/resource_vcd_org_test.go]: Would really like to see consistent handling of timeouts in the adapter. Would you think about it and file a separate issue for this? It should lay out the idea how can we resolve this, like what configurable parameter we need and which places in the code need to be refactored.

File: website/docs/r/org.html.markdown [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+## Example Usage
+
+```
+resource "vcd_org" "my-org" {
+  name             = "my-org"
+  full_name        = "My organization"
+  description      = "The pride of my work"
+  is_enabled       = "true"
+  delete_force     = "true"
+  delete_recursive = "true"
```
Comment (lvirbalas) [inline website/docs/r/org.html.markdown]: Please switch places of delete_force and delete_recursive parameters to better match the logical relationship between them:

```
delete_recursive = "true"
delete_force     = "true"  
```

File: website/docs/r/org.html.markdown [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Org name
+* `full_name` - (Required) Org full name
+* `delete_recursive` - (Required) - pass `delete_recursive`=true as query parameter to remove an organization or VDC and any objects it contains that are in a state that normally allows removal.
```
Comment (lvirbalas) [inline website/docs/r/org.html.markdown]: Strange code quoting. What about this (include the true/false assignment in the code quote):
`delete_recursive=true`

File: website/docs/r/org.html.markdown [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+* `delete_force` - (Required) - pass `delete_force`=true and `delete_recursive`=true to remove an organization or VDC and any objects it contains, regardless of their state.
+* `is_enabled` - (Optional) - True if this organization is enabled (allows login and all other operations). Default is `false`.
+* `description` - (Optional) - Org description. There is no default value.
```
Comment (lvirbalas) [inline website/docs/r/org.html.markdown]: "There is no default value" -> "Default is empty"?

File: website/docs/r/org.html.markdown [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+Provides a vCloud Director Organization resource. This can be used to create and delete an organization.
+---
+
+# vcd\_org
+
+Provides a vCloud Director Org resource. This can be used to create and delete an organization.
+
+Supported in provider *v2.0+*
+
+## Example Usage
```
Comment (lvirbalas) [inline website/docs/r/org.html.markdown]: I believe creation of an Org is only possible with the sys admin user (correct me, if you know of a way to do it with other user type). Hence, we need a sentence about this and it wouldn't hurt to add a `provider` sample in the example below with the correct user.

File: vcd/resource_vcd_org.go (hunk 3 above)
Comment (dataclouder) [reply vcd/resource_vcd_org.go]: Originally, this was a mandatory property.  I can either revert it to `Required` or change the default as `true`. Which one would be preferred?

File: vcd/resource_vcd_org_test.go (hunk 5 above)
Comment (dataclouder) [reply vcd/resource_vcd_org_test.go]: I am working on it. But we need some discussion on this matter.

File: vcd/resource_vcd_org.go (hunk 3 above)
Comment (lvirbalas) [reply vcd/resource_vcd_org.go]: Let's make Org enabled by default, which will lead to a usable configuration with less parameters.

File: website/docs/r/org.html.markdown [hunk 10]
```diff
@@ -9,19 +9,26 @@ Provides a vCloud Director Organization resource. This can be used to create and
 # vcd\_org
 
 Provides a vCloud Director Org resource. This can be used to create and delete an organization.
+Requires administrator privileges.
```
Comment (lvirbalas) [inline website/docs/r/org.html.markdown]: For clarity I would use "system administrator", as org admin is also administrator.

Comment (dataclouder) [reply website/docs/r/org.html.markdown]: done

**Pull request merged**